### PR TITLE
Ability to pass the regional format in addition of the language (issues #4 and #62)

### PR DESCRIPTION
**Pull request by jilouc**
It can be very useful for some languages like Chinese.
As mentioned in issues #4 and #62, sometimes the language alone isn't sufficient.

Simplified Chinese (cmn-Hans) is used both in China and Hong-Kong for example.
If you pass `AppleLanguages (cmn-Hans)` and `AppleLocale cmn-Hans`, and call

```
[[NSLocale currentLocale] objectForKey:NSLocaleCurrencySymbol]
```

the result is undefined, you get `¤`
Using `cmn-Hans_CN` for `AppleLocale`, you get the Yuan symbol `￥` and for `cmn-Hans_HK`, you get the dollar symbol `$`.

I added the ability to specify a [language, locale] pair in the languages list of the Snapfile.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -195,6 +195,16 @@ languages([
 ])
 ```
 
+If you want to use a specific locale, you can also use
```
Comment (KrauseFx) [inline README.md]: Could you append this directly to the already existing `languages` section of the README? https://github.com/KrauseFx/snapshot#languages

File: lib/snapshot/runner.rb [hunk 2]
```diff
@@ -17,13 +17,21 @@ def work(clean: true, build: true)
       FileUtils.rm_rf SnapshotConfig.shared_instance.screenshots_path if SnapshotConfig.shared_instance.clear_previous_screenshots
 
       SnapshotConfig.shared_instance.devices.each do |device|
-        SnapshotConfig.shared_instance.languages.each do |language|
-          reinstall_app(device, language) unless ENV["SNAPSHOT_SKIP_UNINSTALL"]
+        SnapshotConfig.shared_instance.languages.each do |language_item|
+
+          if language_item.instance_of? String
```
Comment (KrauseFx) [inline lib/snapshot/runner.rb]: Could you drop the space after the `?` for consistence?

**Comment by KrauseFx** [comment]
Looks great, thanks for submitting the pull request and working on this :+1: 

I attached 2 comments. I assume you've tested it already on your computer?

**Comment by jilouc** [comment]
Yes, I've already tested it since I needed it to submit my app yesterday (in which I encountered the Simplified Chinese issue). 
Thanks for the incredible amount of work you put into fastlane :)

**Comment by KrauseFx** [comment]
Thank you for the kind words and your contribution to `snapshot` @jilouc, really appreciated :+1:

**Comment by artem-alexandrov** [comment]
I don't know if this is a new issue, or some of already created, but I have next problem:
If i write "en-US", "it-IT" and so on (with second part), language doesn't change on the simulator. I've tried to write just "en", "it" and it worked, but next I wanted to submit metadata to the appstore with deliver, and got an error that folders with screenshots should be named WITH second part.. it's not really convenient to rename them each time... please help me get this done right!

**Comment by KrauseFx** [comment]
@artem-alexandrov I don't really see the problem. Take a look at a working [example configuration](https://github.com/fastlane/examples/blob/master/MindNode/Snapfile).

**Comment by artem-alexandrov** [comment]
@KrauseFx I've tried just like this and the language doesn't change :( It's english for all settings

**Comment by KrauseFx** [comment]
@artem-alexandrov if you can provide me an empty example project with a configuration that runs into this problem? It's working fine for my projects.

**Comment by DominicFrei** [comment]
@KrauseFx I think I can reproduce @artem-alexandrov's problem. At least it sounds like mine. Unfortunately I can only provide an example, but no solution ..
https://github.com/DominicFrei/FastlaneExample

It's just an empty single application template with a UITextField. My problem to be exactly is:
When I call 'snapshot' with two locales defined in the Snapfile, the localisation of text works (not included in that example), but the iOS simulator itself (identified by the status bar in the example screenshots) and the keyboard (that's my main problem) remain English.
[Screenshots](https://github.com/DominicFrei/FastlaneExample/tree/master/fastlane/screenshots/de-DE) are committed as well and can be found in the example repo.

It sounds much like that [Xcode 6.1 bug reported at Stackoverflow](http://stackoverflow.com/questions/26146668/ios8-1-simulator-always-uses-us-keyboard-layout-despite-german-hardware-keyboard), which was also mentioned in [#4](https://github.com/KrauseFx/snapshot/issues/4), except for the fact that I'm using Xcode 8.3 (and the 8.3 iOS simulator).

**Comment by KrauseFx** [comment]
That's true, it seems like only the app is launched in the locale, but not the simulator itself. I'm using a clean status bar (linked in the README) everywhere.

**Comment by DominicFrei** [comment]
@KrauseFx The status bar is not really a problem. As soon as I use frameit and modify my screenshots furthermore, it will disappear anyway, as you already suggested. Thanks for that! :)

It's more about the keyboard which is shown in some screenshots and I want to present a perfect experience to the user (hence a German keyboard in a German screenshot).

I tried to alter the `AppleLanguages` and `AppleLocale` in that `instruments` command you use in `snapshot`:
`instruments -w 'iPhone 6 (8.3 Simulator)' -D '/tmp/snapshot_traces/trace' -t 'Automation' /tmp/snapshot/build/iDecide4u.app -e UIARESULTSPATH '/tmp/snapshot_traces' -e UIASCRIPT '/Users/dominicfrei/Library/Mobile Documents/com~apple~CloudDocs/repositories/iDecide4u/fastlane/snapshot.js' -AppleLanguages '(de-DE)' -AppleLocale 'de-DE'`

I also changed the `Application Language` and `Application Region` in the `scheme`, which did not have any effect on the keyboard, too.

Further investigations will be made by me as well. Hopefully I'll find a solution and can provide it to the project.

**Comment by KrauseFx** [comment]
Great, please keep me up to date @DominicFrei :+1:

**Comment by KrauseFx** [comment]
I like how you want to make your screenshots even more perfect :+1:

**Comment by DominicFrei** [comment]
Another tool I found to change the simulator language is [ios-sim-locale](https://code.google.com/p/ios-sim-locale/), but it's no longer maintained.

Nevertheless, there is [an issue](https://code.google.com/p/ios-sim-locale/issues/detail?id=2) where they talk about about manipulating the iOS simulator locale, language and keyboard directly in the simulator's preferences, which can be found at 
`<home>/Library/Developer/CoreSimulator/Devices/<device_id>/data/Library/Preferences/.GlobalPreferences.plist`

The device id's can be found via `xcrun simctl list`.

In the resulting `.plist` there are three entries that are important:
* AppleLocale
* AppleLanguages
* AppleKeyboards

They can be manipulated via [PlistBuddy](https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man8/PlistBuddy.8.html).

And in the end, I had a localised simulator with a localised status bar and a localised keyboard and localised texts. Exactly what I wanted: the perfect screenshot. ;)

**Comment by KrauseFx** [comment]
Update on this issue: `snapshot` will use the new UI Tests in the future instead of the `instruments` command. It would be a good idea to look in that direction :+1:

**Pull request merged**